pkg/state/invoke_applier.go

```diff
@@ -339,8 +339,8 @@ func (ia *invokeApplier) fallibleValidation(tx proto.Transaction, info *addlInvo
 		KeySizeValidationVersion: keySizeValidationVersion,
 		MaxDataEntriesSize:       maxDataEntriesSize,
 	}
-
-	if err := proto.ValidateActions(info.actions, restrictions, int(info.libVersion)); err != nil {
+	validatePayments := info.checkerInfo.height > ia.settings.InternalInvokePaymentsValidationAfterHeight
```

Comment:
Don't you think that we have to validate payments before `the hight` and after it? It seems to me that this code makes validation work only after `ia.settings.InternalInvokePaymentsValidationAfterHeight`. Maybe we should do something like this:
```
validatePayments := !(info.checkerInfo.height == ia.settings.InternalInvokePaymentsValidationAfterHeight)
```
Or is it not related to the last problem?

Reply:
I realized that before `the height` the payments has not been validated by the main Scala node, so they must be validated only after `the height`